design/behavior.actions.md

```diff
@@ -0,0 +1,94 @@
+# Behavior Actions
+
+Behaviors are high level concepts that are made up of one or more actions.  These actions make up the interactions
+that the user makes with the given behavior.  These actions are broken up in to three states: `started` and `stopped`.
+
+- `started` - Fired once when the action is first beginning to be performed.
+- `stopped` - Fired once when the action is stopped being performed.
+
+**Proposed**: In addition, there is an optional update event that can be registered to the action for getting updates while that
+action is being performed on the client.
+
+## Architecture
+
+### Action State
+``` ts
+export type ActionState
```

Comment:
Are these states or state transitions, a.k.a. events? They seem to be used as both, which is confusing to me.

Reply:
These are state events.  

- `started` is an event raised when the action is first started
- `performing` is an optional update event that happens while the action is being performed
- `stopped` is an event raised when the action is stopped.